pkg/client/client.go

```diff
@@ -0,0 +1,163 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package client implements image factory HTTP API client.
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/siderolabs/gen/xerrors"
+
+	"github.com/siderolabs/image-factory/pkg/schematic"
+)
+
+// InvalidSchematicError is parsed from 400 response from the server.
+type InvalidSchematicError struct {
+	Details string
+}
+
+// Error implements error interface.
+func (e *InvalidSchematicError) Error() string {
+	return fmt.Sprintf("invalid schematic: %s", e.Details)
+}
+
+// IsInvalidSchematicError checks if the error is invalid schematic.
+func IsInvalidSchematicError(err error) bool {
+	return xerrors.TypeIs[*InvalidSchematicError](err)
+}
+
+// ExtensionInfo defines extensions versions list response item.
+type ExtensionInfo struct {
+	Name   string `json:"name"`
+	Ref    string `json:"ref"`
+	Digest string `json:"digest"`
+}
+
+// Client is the Image Factory HTTP API client.
+type Client struct {
+	address string
+	client  http.Client
+}
+
+// New creates a new Omni API client.
+func New(address string, timeout time.Duration) *Client {
```

Comment:
it feels it's not `address`, but rather base URL, and we could parse it here to make sure it's valid, and use `.Join` method to append patch to it?

Reply:
I do join path in each method. Should be enough I guess

Reply:
ok, updated